docs/source/cpp/compute.rst

```diff
@@ -812,45 +812,47 @@ The third set of functions examines string elements on a byte-per-byte basis:
++-------------------------+--------+-----------------------------------------+------------------------+-----------------------------------+-------+
+| ascii_lower             | Unary  | String-like                             | String-like            |                                   | \(1)  |
++-------------------------+--------+-----------------------------------------+------------------------+-----------------------------------+-------+
+| ascii_reverse           | Unary  | String-like                             | String-like            |                                   | \(2)  |
++-------------------------+--------+-----------------------------------------+------------------------+-----------------------------------+-------+
+| ascii_swapcase          | Unary  | String-like                             | String-like            |                                   | \(1)  |
++-------------------------+--------+-----------------------------------------+------------------------+-----------------------------------+-------+
+| ascii_title             | Unary  | String-like                             | String-like            |                                   | \(1)  |
++-------------------------+--------+-----------------------------------------+------------------------+-----------------------------------+-------+
+| ascii_upper             | Unary  | String-like                             | String-like            |                                   | \(1)  |
++-------------------------+--------+-----------------------------------------+------------------------+-----------------------------------+-------+
+| binary_length           | Unary  | Binary- or String-like                  | Int32 or Int64         |                                   | \(3)  |
++-------------------------+--------+-----------------------------------------+------------------------+-----------------------------------+-------+
+| binary_replace_slice    | Unary  | String-like                             | Binary- or String-like | :struct:`ReplaceSliceOptions`     | \(4)  |
++-------------------------+--------+-----------------------------------------+------------------------+-----------------------------------+-------+
+| replace_substring       | Unary  | String-like                             | String-like            | :struct:`ReplaceSubstringOptions` | \(5)  |
++-------------------------+--------+-----------------------------------------+------------------------+-----------------------------------+-------+
+| replace_substring_regex | Unary  | String-like                             | String-like            | :struct:`ReplaceSubstringOptions` | \(6)  |
++-------------------------+--------+-----------------------------------------+------------------------+-----------------------------------+-------+
+| string_repeat           | Binary | Binary/String (Arg 0); Integral (Arg 1) | Binary- or String-like |                                   | \(7)  |
```

Comment:
Since there's already "binary_replace_slice", should it be called "binary_repeat" instead?

Reply:
Well, [from a previous discussion](https://ursalabs.zulipchat.com/#narrow/stream/180245-dev/topic/Stringlike.20kernels.20on.20binary.20data), I am following the pattern that a name with `string` expects/supports both binary and string encoded data. While the `binary` prefix only expects binary non-encoded data and `ascii/utf8` are for encoding-specific functions.
There are two solutions to be consistent with functions that have either a `binary` or `string` prefix:
1. Change them all to `binary`
    * `string_repeat` --> `binary_repeat`
    * `string_is_ascii` --> `binary_is_ascii`
2. Change them all to `string` as they seem to support both binary/string types
    * `binary_length` --> `string_length`
    * `binary_replace_slice` --> `string_replace_slice`
    * `binary_join` --> `string_join`
    * `binary_join_element_wise` --> `string_join_element_wise`

Reply:
I think solution 2 is better.

Reply:
Both solutions are fine to me.